ui/store/actions.ts

```diff
@@ -3282,6 +3282,45 @@ export function setParticipateInMetaMetrics(
   };
 }
 
+export function setDataCollectionForMarketing(
+  dataCollectionPreference: boolean,
+): ThunkAction<
+  Promise<[boolean, string]>,
+  MetaMaskReduxState,
+  unknown,
+  AnyAction
+> {
+  return (dispatch: MetaMaskReduxDispatch) => {
+    log.debug(`background.setDataCollectionForMarketing`);
+    return new Promise((resolve, reject) => {
+      callBackgroundMethod<string>(
+        'setDataCollectionForMarketing',
+        [dataCollectionPreference],
+        (err, metaMetricsId) => {
+          log.debug(err);
+          if (err) {
+            dispatch(displayWarning(err));
```

Comment:
`displayWarning` is deprecated, please find some other way of handling this error. This method will display the error totally out of context in random places in the app, and it'll stick around for a seemingly arbitrary amount of time. Bad UX.

Once this is converted to use `submitRequestToBackground`, this error variable won't be here unless we add a `try...catch` block to catch it. We may or may not want to do that, depending on when we expect this to fail.

If it should never fail, I'd recommend not catching the error and letting it bubble up to the global error handler to be captured by Sentry.

If we'd expect it to fail sometimes, then we should catch it. If the failure condition is relevant to the user (e.g. if an operation failed and the user should be informed), then we'll have to think about where in the UI to best display this error. Maybe easiest to throw here and catch it in a UI component where we can show the error.

If we expect it to fail sometimes in a way that isn't important to a user, then we can just log the error to the console and move on.

Reply:
Fixed